aws_lambda_fsm/_pkg_meta.py

```diff
@@ -12,5 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# system imports
+from builtins import map
```

Comment:
Do you need `from builtins import str` here?

Reply:
It winds up in `setup.py`, which also has a number of other string literals that are likewise interpreted as bytestrings in Python 2 & Unicode strings in Python 3. The API there has to be working either way, otherwise we would be getting build problems.

Reply:
@macleodbroad-wf is right. `sdist` works, but `pip install` does not.

```
(py2tst) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$ pip install dist/aws-lambda-fsm-0.19.0.tar.gz
Looking in indexes: https://pypi.workiva.net/simple/, https://pypi.python.org/simple/
Processing ./dist/aws-lambda-fsm-0.19.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-E28Ahh/setup.py", line 54, in <module>
        version=get_version(),
      File "/private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-E28Ahh/setup.py", line 21, in get_version
        pkg_meta = imp.load_source('_pkg_meta', 'aws_lambda_fsm/_pkg_meta.py')
      File "aws_lambda_fsm/_pkg_meta.py", line 16, in <module>
        from builtins import map
    ImportError: No module named builtins

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-E28Ahh/
(py2tst) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$
```

Reply:
He was suggesting that I add another `builtins` import, but the error here is an import error for the *existing* `builtins` import — not recognizing that module at all.
I'll look into it.

Reply:
On a new Python 2.7 virtualenv, I tried `python setup.py sdist` and then `pip install` of the zip. I didn't get the above error.

The package version was 0.21 (now 0.22 after the merge I just did), so it looks like you were trying an old release. Give it a try again with the latest code.

Reply:
```
(foobar) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$ pip install dist/aws-lambda-fsm-0.22.0.tar.gz
Looking in indexes: https://pypi.workiva.net/simple/, https://pypi.python.org/simple/
Processing ./dist/aws-lambda-fsm-0.22.0.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-AYAqQh/setup.py", line 54, in <module>
        version=get_version(),
      File "/private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-AYAqQh/setup.py", line 21, in get_version
        pkg_meta = imp.load_source('_pkg_meta', 'aws_lambda_fsm/_pkg_meta.py')
      File "aws_lambda_fsm/_pkg_meta.py", line 16, in <module>
        from builtins import map
    ImportError: No module named builtins

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/_3/2k6zbxys5j7g0qkm5qn4zgm80000gm/T/pip-req-build-AYAqQh/
(foobar) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$ pip install future
]Looking in indexes: https://pypi.workiva.net/simple/, https://pypi.python.org/simple/
Collecting future
  Downloading https://files.pythonhosted.org/packages/90/52/e20466b85000a181e1e144fd8305caf2cf475e2f9674e797b222f8105f5f/future-0.17.1.tar.gz (829kB)
    100% |████████████████████████████████| 829kB 3.0MB/s
Building wheels for collected packages: future
  Running setup.py bdist_wheel for future ... done
  Stored in directory: /Users/shawnrusaw/Library/Caches/pip/wheels/0c/61/d2/d6b7317325828fbb39ee6ad559dbe4664d0896da4721bf379e
Successfully built future
Installing collected packages: future
Successfully installed future-0.17.1
(foobar) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$ pip install dist/aws-lambda-fsm-0.22.0.tar.gz
Looking in indexes: https://pypi.workiva.net/simple/, https://pypi.python.org/simple/
Processing ./dist/aws-lambda-fsm-0.22.0.tar.gz
Collecting pyyaml==3.11 (from aws-lambda-fsm==0.22.0)
Collecting future==0.16 (from aws-lambda-fsm==0.22.0)
Building wheels for collected packages: aws-lambda-fsm
  Running setup.py bdist_wheel for aws-lambda-fsm ... done
  Stored in directory: /Users/shawnrusaw/Library/Caches/pip/wheels/e7/89/19/09824d23353e2806cfa5bda944c7eaaf74d4e84a69db69d381
Successfully built aws-lambda-fsm
Installing collected packages: pyyaml, future, aws-lambda-fsm
  Found existing installation: future 0.17.1
    Uninstalling future-0.17.1:
      Successfully uninstalled future-0.17.1
Successfully installed aws-lambda-fsm-0.22.0 future-0.16.0 pyyaml-3.11
(foobar) [shawnrusaw@wf14381 aws-lambda-fsm-workflows (python_3)]$
```

i have to pip install future before setup.py will run

Reply:
OK, I was able to reproduce this time — my virtualenv on last try must not have been clean.
I removed the import of the `builtins` version of `map` in `_pkg_meta.py`, should work now.